### DISCUSSION TITLE
What's the difference between the `min_id` and `since_id` on timeline API docs?

### DISCUSSION THREAD

**Discussion by markcellus** [mastodon/mastodon#20223] [Q&A] [answered]
The [API documentation for timelines](https://docs.joinmastodon.org/methods/timelines/) says `min_id` "returns results immediately newer than this id" and `since_id` "returns results newer than this id". But they both sound like the same thing. What does "immediately newer" mean in this context? Can anyone clarify? Thanks for any information.

**Comment by garthk** [comment]
From the Python API client docs:

> All endpoints that are paginated have four parameters: since_id, max_id, min_id and limit. since_id allows you to specify the smallest id you want in the returned data, but you will still always get the newest data, so if there are too many statuses between the newest one and since_id, some will not be returned. min_id, on the other hand, gives you statuses with that minimum id and newer, starting at the given id. max_id, similarly, allows you to specify the largest id you want. By specifying either min_id or max_id (generally, only one, not both, though specifying both is supported starting with Mastodon version 3.3.0) of them you can go through pages forwards and backwards.

**Comment by trwnh** [comment, marked as answer]
max_id sets an upper bound. since_id sets a lower bound. min_id will actually move the cursor back to that point, paginating from there.

example:

- `?since_id=1` will return the latest statuses, since they are all greater than `1`
- `?min_id=1` will return the oldest statuses starting from `1` and going forward

clarified in https://github.com/mastodon/documentation/pull/1209
